dipy/tracking/tests/test_streamline.py

```diff
@@ -1184,5 +1261,118 @@ def test_cluster_confidence():
+
+    # And their weights should be halved:
+    npt.assert_almost_equal(w[0], w[1] / 2)
+    npt.assert_almost_equal(w[-1], w[2] / 2)
+
+    # Test the situation where all the streamlines have an identical node:
+    arr1 = np.array([x, y, z]).T + 2
+    arr2 = np.array([x, y, z]).T + 1
+    arr3 = np.array([x, y, z]).T - 1
+    arr4 = np.array([x, y, z]).T - 2
+
+    arr1[0] = np.array([1, 1, 1])
+    arr2[0] = np.array([1, 1, 1])
+    arr3[0] = np.array([1, 1, 1])
+    arr4[0] = np.array([1, 1, 1])
+
+    bundle_w_id_node = Streamlines([arr1, arr2, arr3, arr4])
+    w = gaussian_weights(Streamlines(bundle_w_id_node), n_points=10)
+    # For this case, the result should be a weight of 1/n_streamlines in that
+    # node for all streamlines:
+    npt.assert_equal(w[:, 0],
+                     np.ones(len(bundle_w_id_node)) * 1/len(bundle_w_id_node))
+
+
+    # Test the situation where all the streamlines are copies of each other:
+    bundle_w_copies = Streamlines([bundle[0], bundle[0], bundle[0], bundle[0]])
+    w = gaussian_weights(bundle_w_copies, n_points=10)
+    # In this case, the entire array should be equal to 1/n_streamlines:
+    npt.assert_equal(w,
+                     np.ones(w.shape) * 1/len(bundle_w_id_node))
+
+    # Test with bundle of length 1:
+    bundle_len_1 = Streamlines([bundle[0]])
+    w = gaussian_weights(bundle_len_1, n_points=10)
+    npt.assert_equal(w, np.ones(w.shape))
+
+    bundle_len_1 = Streamlines([bundle[0]])
+    w = gaussian_weights(bundle_len_1, n_points=10, return_mahalnobis=True)
+    npt.assert_equal(w, np.ones(w.shape) * np.nan)
+
+
+def test_bundle_profile():
+    data = np.ones((10, 10, 10))
+    bundle = Streamlines([np.array([[0, 0., 0],
+                                    [1, 0., 0.],
+                                    [2, 0., 0.]]),
+                          np.array([[0, 0., 0.],
+                                    [1, 0., 0],
+                                    [2, 0,  0.]])])
+
+    profile = bundle_profile(data, bundle)
+    npt.assert_equal(profile, np.ones(100))
+
+    profile = bundle_profile(data, bundle, affine=None, n_points=10,
+                             weights=None)
+    npt.assert_equal(profile, np.ones(10))
+
+    profile = bundle_profile(data, bundle, affine=None, n_points=10,
+                             weights=None)
+    npt.assert_equal(profile, np.ones(10))
+
+    profile = bundle_profile(data, bundle, affine=None, n_points=10,
+                             weights=np.ones((2, 10)) * 0.5)
+    npt.assert_equal(profile, np.ones(10))
+
+
+    # Disallow setting weights that don't sum to 1 across fibers/nodes:
+    assert_raises(ValueError, bundle_profile,
+                  data, bundle, affine=None,
+                  n_points=10, weights=np.ones((2, 10)) * 0.6)
+
+    # Test using an affine:
+    affine = np.eye(4)
+    affine[:, 3] = [-1, 100, -20, 1]
+    # Transform the streamlines:
+    x_bundle = Streamlines([sl + affine[:3, 3] for sl in bundle])
```

Comment:
Same here. Use extend. Or even better transform the data directly using
x_bundle._data